app/js/app.js

```diff
@@ -0,0 +1,106 @@
+//responsive canvas
+$(document).ready(function() {
+  //get canvas and context
+  var c = $('#juice-canvas');
+  var ctx = c.get(0).getContext('2d');
+  var container = $(c).parent();
+  //when browser resizes
+  $(window).resize(responsiveCanvas);
+  function responsiveCanvas() {
+    c.attr('width', $(container).width()); //max width
+    c.attr('height', $(container).height()); //max height
+  }
+  responsiveCanvas();
+});
+
+//mobile and desktop
+var c = document.getElementById('juice-canvas');
+var ctx = c.getContext('2d');
+var isDrawing = false;
+ctx.strokeStyle = '#333';
+ctx.lineWidth = 2;
+//touch
+c.addEventListener('touchstart', function(e) {
+  var rect = c.getBoundingClientRect();
+  var x = Math.round((e.clientX - rect.left) / (rect.right - rect.left) * c.width);
+  var y = Math.round((e.clientY - rect.top) / (rect.bottom - rect.top) * c.height);
+  var touch = e.touches[0];
+  var mouseEvent = new MouseEvent('mousedown', {
+    clientX: touch.clientX,
+    clientY: touch.clientY
+  });
+  c.dispatchEvent(mouseEvent);
+}, false);
+
+c.addEventListener('touchmove', function(e) {
+  var touch = e.touches[0];
+  var mouseEvent = new MouseEvent('mousemove', {
+    clientX: touch.clientX,
+    clientY: touch.clientY
+  });
+  c.dispatchEvent(mouseEvent);
+}, false);
+
+c.addEventListener('touchend', function(e) {
+  var mouseEvent = new MouseEvent('mouseup', {});
+  c.dispatchEvent(mouseEvent);
+}, false);
+//mouse
+c.onmousedown = function(e) {
+  e.preventDefault();
+  var rect = c.getBoundingClientRect();
+  var x = Math.round((e.clientX - rect.left) / (rect.right - rect.left) * c.width);
+  var y = Math.round((e.clientY - rect.top) / (rect.bottom - rect.top) * c.height);
+  isDrawing = true;
+  ctx.lineJoin = ctx.lineCap = 'round';
+  ctx.moveTo(x,y);
+}
+
+c.onmousemove = function(e) {
+  e.preventDefault();
+  var rect = c.getBoundingClientRect();
+  var x = Math.round((e.clientX - rect.left) / (rect.right - rect.left) * c.width);
+  var y = Math.round((e.clientY - rect.top) / (rect.bottom - rect.top) * c.height);
+  if(isDrawing) {
+    ctx.lineTo(x,y);
+    ctx.stroke();
+  }
+}
+
+c.onmouseup = function(e) {
+  e.preventDefault();
+  isDrawing = false;
+} 
+
+//timer
+function getTimeRemaining(endtime) {
+  var t = Date.parse(endtime) - Date.parse(new Date());
+  var seconds = Math.floor((t / 1000) % 60);
+  var minutes = Math.floor((t / 1000 / 60) % 60);
+  return {
+    'total': t,
+    'minutes': minutes,
+    'seconds': seconds
+  };
+}
+
+function initializeClock(id, endtime) {
+  var clock = document.getElementById('timer');
+  var minutesSpan = clock.querySelector('.minutes');
+  var secondsSpan = clock.querySelector('.seconds');
```

Comment:
The clock still works without those declarations, but `initializeClock` sets the start time in the span tags for minutes and seconds.
I've been playing around with the placement of `updateClock` and the timer doesn't initialize or countdown unless `updateClock` is nested in `initializeClock`. It's grabbing the `endtime` from `initializeClock` but any other placement breaks it... I'll keep playing with it `initializeClock`